ibis/client.py

```diff
@@ -378,6 +380,22 @@ def _get_list(self, cur, i=0):
         else:
             return []
 
+    def _get_udfs(self, cur):
+        tuples = cur.fetchall()
+        if len(tuples) > 0:
+            regex = re.compile("(.*?)\s*\((.*?)\)")
+            result = []
+            for out_type, sig in tuples:
+                reg_result = regex.search(sig)
+                name = reg_result.group(1)
+                in_types = [x.lower() for x in reg_result.group(2).split(', ')]
+                inputs = [udf._scalar_conversion_types[x] for x in in_types]
+                output = udf._scalar_conversion_types[out_type.lower()]
+                result.append(udf.UDFInfo(inputs, output, name))
```

Comment:
@wesm 
If we want friendlier show_function output so people can easily drop functions, we could not convert from strings to Ibis types. I'll flesh out some potential user workflows so you get a better picture.